libs/langchain-mistralai/src/chat_models.ts

```diff
@@ -0,0 +1,410 @@
+  >
+  extends SimpleChatModel<CallOptions>
+  implements ChatMistralAIInput
+{
+  // Used for tracing, replace with the same name as your class
+  static lc_name() {
+    return "ChatMistralAI";
+  }
+
+  modelName = "mistral-tiny";
+
+  client = new MistralClient();
+
+  temperature = 0.7;
+
+  streaming = false;
+
+  topP = 1;
+
+  maxTokens: number;
+
+  safeMode = false;
+
+  randomSeed?: number;
+
+  lc_serializable = true;
+
+  constructor(fields?: ChatMistralAIInput) {
+    super(fields ?? {});
+    const apiKey = fields?.apiKey ?? getEnvironmentVariable("MISTRAL_API_KEY");
+    if (!apiKey) {
+      throw new Error(
+        "API key MISTRAL_API_KEY is missing for MistralAI, but it is required."
+      );
+    }
+    this.client = new MistralClient(apiKey, fields?.endpoint);
+  }
+
+  _llmType() {
+    return "chat_mistral_ai";
+  }
+
+  /**
+   * For some given input messages and options, return a string output.
+   * @param {Array<BaseMessage>} messages The messages to send to the model.
+   * @param {this["ParsedCallOptions"]} options The options to use when calling the model.
+   * @param {CallbackManagerForLLMRun | undefined} runManager Optional, the callback manager to use for this run.
+   * @returns {Promise<string>} The response from the model.
+   */
+  async _call(
+    messages: BaseMessage[],
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): Promise<string> {
+    const chatResponse = await this._generate(messages, options, runManager);
+    const { content } = chatResponse.generations[0].message;
+    if (typeof content === "string") {
+      return content;
+    }
+    throw new Error(
+      `ChatMistralAI does not support non text message content. Received: ${JSON.stringify(
+        content,
+        null,
+        2
+      )}`
+    );
+  }
+
+  /**
+   * Get the parameters used to invoke the model
+   */
+  invocationParams(): Omit<MistralAIChatCompletionOptions, "messages"> {
+    const params = {
+      model: this.modelName,
+      temperature: this.temperature,
+      topP: this.topP,
+      maxTokens: this.maxTokens,
+      safeMode: this.safeMode,
+      randomSeed: this.randomSeed
+    };
+    return params;
+  }
+
+  /**
+   * Calls the MistralAI API with retry logic in case of failures.
+   * @param {MistralAIChatCompletionOptions} input The input to send to the MistralAI API.
+   * @returns {Promise<MistralAIChatCompletionResult | AsyncGenerator<MistralAIChatCompletionResult>>} The response from the MistralAI API.
+   */
+  async completionWithRetry(
+    input: MistralAIChatCompletionOptions,
+    streaming: true
+  ): Promise<AsyncGenerator<MistralAIChatCompletionResult>>;
+
+  async completionWithRetry(
+    input: MistralAIChatCompletionOptions,
+    streaming: false
+  ): Promise<MistralAIChatCompletionResult>;
+
+  async completionWithRetry(
+    input: MistralAIChatCompletionOptions,
+    // Hack to satisfy TS return type
+    streaming: boolean = this.streaming
```

Comment:
I think we solve this in other ways elsewhere?

Reply:
Outdated comment, removed. But I still need to pass in a `streaming` param because you can call `.stream()` without `this.streaming` being true.

Reply:
Elsewhere if you call `.stream()` it will override any `this.streaming` property set on the instance and return the most chunks possible (which is what users want)